constants.php

```diff
@@ -1,35 +1,64 @@
 <?php
+//NB: Do not edit; use ./constants.local.php instead.
+
+//<Not customisable>
 define('FRESHRSS_VERSION', '1.8.1-dev');
 define('FRESHRSS_WEBSITE', 'https://freshrss.org');
 define('FRESHRSS_WIKI', 'https://freshrss.github.io/FreshRSS/');
 
-define('FRESHRSS_USERAGENT', 'FreshRSS/' . FRESHRSS_VERSION . ' (' . PHP_OS . '; ' . FRESHRSS_WEBSITE . ')');
-
-// PHP text output compression http://php.net/ob_gzhandler (better to do it at Web server level)
-define('PHP_COMPRESSION', false);
-
-// Maximum log file size in Bytes, before it will be divided by two
-define('MAX_LOG_SIZE', 1048576);
+define('FRESHRSS_PATH', __DIR__);
+define('PUBLIC_PATH', FRESHRSS_PATH . '/p');
+define('PUBLIC_TO_INDEX_PATH', '/i');
+define('INDEX_PATH', PUBLIC_PATH . PUBLIC_TO_INDEX_PATH);
+define('PUBLIC_RELATIVE', '..');
+define('LIB_PATH', FRESHRSS_PATH . '/lib');
+define('APP_PATH', FRESHRSS_PATH . '/app');
+define('EXTENSIONS_PATH', FRESHRSS_PATH . '/extensions');
+//</Not customisable>
 
-// Constantes de chemins
-define('FRESHRSS_PATH', dirname(__FILE__));
-
-	define('PUBLIC_PATH', FRESHRSS_PATH . '/p');
-		define('PUBLIC_TO_INDEX_PATH', '/i');
-		define('INDEX_PATH', PUBLIC_PATH . PUBLIC_TO_INDEX_PATH);
-		define('PUBLIC_RELATIVE', '..');
+if (file_exists(__DIR__ . '/constants.local.php')) {
```

Comment:
Would it be clearer to call it something like `constants.persistent.php` instead?

Reply:
I hesitated a bit on the naming. I wanted to make it clear that it was something for local administrators and not from a default FreshRSS configuration. "local" is a term used in Linux FHS as well as Windows for related needs.

Reply:
But my FreshRSS itself is already local, conceptually speaking. :-P

How does Windows use it?

Reply:
I was referring to https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard , where *local* means *specific to this host*, which is a bit what I wanted to achieve there.
In Windows, there is *%LOCALAPPDATA%* for a similar usage.

Reply:
I meant conceptually it's like a script in `/usr/local` even if I personally happen to store it in `/var/someplace`.

I suppose the distinction is somewhat similar — I assume that in Windows you have settings provided over the network or from a system default, and in %LOCALAPPDATA% a Windows program might store things like recently opened files and preferences.

But I'm not sure if it clarifies or obscures. If you already know that your user profile stores stuff in `~/.local` or `%USERPROFILE%\Local Settings` it's fine, but if you don't it's somewhat mystifying. That's why I suggested an imo more self-explanatory word like `persistent`.

Reply:
While I see what you mean with *persistent*, I am unsure it will be well understood as something users can edit, and it does not seem to be so commonly used. In any case, it would be nice to add some documentation somewhere :-)